src/common/com/intellij/plugins/haxe/ide/info/HaxeParameterDescription.java

```diff
@@ -17,66 +18,80 @@
+import org.jetbrains.annotations.Nullable;
 
-import java.util.ArrayList;
-import java.util.List;
+import javax.xml.transform.Result;
 
-/**
- * @author: Fedor.Korotkov
- */
 public class HaxeParameterDescription {
-  private final String name;
-  private final String typeText;
-  private final String varInitValue;
-
-  public HaxeParameterDescription(String name, String text, String varInitText) {
-    this.name = name;
-    typeText = text;
-    varInitValue = varInitText;
-  }
+  private final boolean isOptional;
+  private final boolean hasInitialValue;
 
+  private final String text;
 
-  public static HaxeParameterDescription[] getParameters(HaxeNamedComponent element, HaxeGenericSpecialization specialization, boolean isExtension) {
-    final HaxeParameterList parameterList = PsiTreeUtil.getChildOfType(element, HaxeParameterList.class);
-    if (parameterList == null) {
-      return new HaxeParameterDescription[0];
-    }
-    List<HaxeParameter> list = parameterList.getParameterList();
-    if (isExtension) {
-      list = new ArrayList<HaxeParameter>(list);
-      list.remove(0);
-    }
-    final HaxeParameterDescription[] result = new HaxeParameterDescription[list.size()];
-    for (int i = 0, size = list.size(); i < size; i++) {
-      HaxeParameter parameter = list.get(i);
-      String typeText = "";
-      if (parameter.getTypeTag() != null) {
-        typeText = HaxePresentableUtil.buildTypeText(element, parameter.getTypeTag(), specialization);
-      }
-      String varInitText = "";
-      HaxeVarInit varInit = parameter.getVarInit();
-      if (varInit != null) {
-        final HaxeExpression varInitExpression = varInit.getExpression();
-        varInitText = varInitExpression == null ? "" : varInitExpression.getText();
-      }
-      result[i] = new HaxeParameterDescription(parameter.getName(), typeText, varInitText);
-    }
-    return result;
+  private final ResultHolder resultHolder;
+
+  HaxeParameterDescription(String name,
+                           String type,
+                           String initialValue,
+                           boolean isOptional,
+                           ResultHolder resultHolder) {
+
+    this.isOptional = isOptional;
+    this.hasInitialValue = initialValue != null;
+
+    this.text = compileDescription(name, type, initialValue);
+    this.resultHolder = resultHolder;
   }
 
   @Override
   public String toString() {
+    return text;
+  }
+
+  public boolean isOptional() {
+    return isOptional;
+  }
+
+  public boolean hasInitialValue() {
+    return hasInitialValue;
+  }
+
+  public boolean isPredefined() {
+    return isOptional || hasInitialValue;
+  }
+
+  private String compileDescription(String name, @Nullable String type, @Nullable String initialValue) {
+    final boolean hasInitialValue = initialValue != null && !initialValue.isEmpty();
+    final boolean hasType = type != null && !type.isEmpty();
+    final boolean isValuePredefined = isOptional || hasInitialValue;
```

Comment:
I agree, naming is not very good, but `||` is proper operator. In the given place predefined value means that parameter is optional (and it has value `null` by default) or it has intial value. So that's the reason.